python/cudnn/sdpa/fwd/api_dsl.py

```
@@ -1016,13 +1077,12 @@ def _execute_thd(self, q_buf, k_buf, v_buf, o_buf, scale_softmax_log2, sinks, se
-            return buf.as_strided((1, t, h, d), (t * h * d, h * d, d, 1), buf.storage_offset())
-
-        Q = _packed(q_buf, t_q, qh, d_qk)
-        K = _packed(k_buf, t_kv, kh, d_qk)
-        V = _packed(v_buf, t_kv, kh, d_v)
-        O = _packed(o_buf, t_q, qh, d_v)
+        # Normalize declared strides to the kernel's packed contract (zero-copy
+        # when already packed; gather/scatter through carved scratch otherwise).
+        Q, _ = self._thd_normalized(q_buf, self.q_desc, t_q, carver, gather=True)
+        K, _ = self._thd_normalized(k_buf, self.k_desc, t_kv, carver, gather=True)
+        V, _ = self._thd_normalized(v_buf, self.v_desc, t_kv, carver, gather=True)
+        O, o_scatter = self._thd_normalized(o_buf, self.o_desc, t_q, carver, gather=False)
```

Ruff (0.16.1)

[error] Ambiguous variable name: `O`

(E741)